components/admin_console/reset_email_modal/reset_email_modal.jsx

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import {adminResetEmail} from 'actions/admin_actions.jsx';
+import * as Utils from 'utils/utils.jsx';
+
+export default class ResetEmailModal extends React.Component {
+    static propTypes = {
+        user: PropTypes.object,
+        show: PropTypes.bool.isRequired,
+        onModalSubmit: PropTypes.func,
+        onModalDismissed: PropTypes.func,
+        passwordConfig: PropTypes.object,
+    };
+
+    static defaultProps = {
+        show: false,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            error: null,
+        };
+    }
+
+    componentWillUnmount() {
+        this.setState({error: null});
+    }
+
+    doSubmit = (e) => {
+        e.preventDefault();
+
+        if (this.refs.email) {
+            const email = this.refs.email.value;
+            if (email === '' || !Utils.isEmail(email)) {
+                const errMsg = (
+                    <FormattedMessage
+                        id='user.settings.general.validEmail'
```

Comment:
@saturninoabril, what's our practice for localizing strings that occur in more than one place? Should we introduce a duplicate translation here for the admin console case, or is it ok to reuse ids from other components like this?